internal/clients/state/v3/state.go

```diff
@@ -107,28 +120,57 @@ func NewClient(ctx context.Context, pbClient pb.PluginClient, tableName string)
 			}
 			keys := record.Columns()[0].(*array.String)
 			values := record.Columns()[1].(*array.String)
+			versions := record.Columns()[2].(*array.Uint64)
 			for i := 0; i < keys.Len(); i++ {
-				c.mem[keys.Value(i)] = values.Value(i)
+				k, val := keys.Value(i), values.Value(i)
+
+				if _, ok := c.mem[k]; !ok {
+					c.mem[k] = make(map[uint64]string)
+				}
+				var ver uint64
+				if versions.IsValid(i) {
+					ver = versions.Value(i)
+				}
+				c.mem[k][ver] = val
```

Comment:
Why do we need to keep all the versions in memory? We should be able to keep only the latest version, along with its version number? 

And maybe in flush we can only write and increment the version by one if the value of the key was updated. Otherwise we might be writing a lot of intermediate values that will never be used

Reply:
done in https://github.com/cloudquery/plugin-sdk/pull/1604/commits/12a143b3ada6d01b24be7982eea814a1e580223e